### ISSUE TITLE
TestIngressController failure

### ISSUE THREAD

**Issue by kyessenov** [istio/istio#4319]
```
=== RUN   TestIngressController
2018-03-16T05:29:23.475295Z	info	Ingress controller watching namespaces "test"
2018-03-16T05:29:23.486293Z	info	ingress event add for test/nginx-ingress
--- FAIL: TestIngressController (3.00s)
	controller_test.go:210: timeout waiting to receive expected rule count 5, actually received 0
```

Test successfully passes on prow in the same PR.

**Comment by xiaolanz** [comment]
should these also moved to e2e? since it's test to apis and get flaky.

**Comment by kyessenov** [comment]
TestServices is also flaking. Should we move both to e2e pipeline? We can institute a ban on using k8s in unit tests, which I would support.

cc @louiscryan @costinm

**Comment by kyessenov** [comment]
cc @nmittler I want to move these tests to e2e in pilot as well, since they are not really unit.

**Comment by xiaolanz** [comment]
attn: #4296

**Comment by cmluciano** [comment]
@xiaolanz Is this fixed by #4510
